File: template/src/main/java/ru/surfstudio/standard/app/dagger/AppModule.java
package ru.surfstudio.standard.app.dagger;

import android.content.Context;

import dagger.Module;
import dagger.Provides;
import ru.surfstudio.android.core.app.dagger.scope.PerApplication;

@Module
public class AppModule {
    private Context appContext;

    public AppModule(Context appContext) {
        this.appContext = appContext;
    }

    @Provides
    @PerApplication
    Context provideContext() {
        return appContext;
    }
}

File: template/src/main/java/ru/surfstudio/standard/app/dagger/AppModule.kt
package ru.surfstudio.standard.app.dagger

import android.content.Context

import dagger.Module
import dagger.Provides
import ru.surfstudio.android.core.app.ActiveActivityHolder
import ru.surfstudio.android.core.app.CoreApp
import ru.surfstudio.android.core.app.StringsProvider
import ru.surfstudio.android.dagger.scope.PerApplication


@Module
class AppModule(private val coreApp: CoreApp) {

    @PerApplication
    @Provides
    internal fun provideActiveActivityHolder(): ActiveActivityHolder {
        return coreApp.activeActivityHolder
    }

    @PerApplication
    @Provides
    internal fun provideContext(): Context {
        return coreApp
    }

    @PerApplication
    @Provides
    internal fun provideStringsProvider(context: Context): StringsProvider {
        return StringsProvider(context)
    }
}
